Lean4Lean/TypeChecker.lean

```diff
@@ -416,9 +475,10 @@ def reduceNat (e : Expr) : RecM (Option Expr) := do
     if f == ``Nat.div then return ← reduceBinNatOp Nat.div a b
     if f == ``Nat.beq then return ← reduceBinNatPred Nat.beq a b
     if f == ``Nat.ble then return ← reduceBinNatPred Nat.ble a b
+  -- FIXME should return some if e is already a nat literal?
```

Comment:
no, this returns `some` if it performed some reduction, so nat literals return `none`.